### PR TITLE
Fix compiler warnings with -Wextra

### PR DESCRIPTION
**Pull request by meyerj**
... about misleading indentation, comparisons between signed and unsigned types and an extra semicolon.

### CODE REVIEW

File: orocos_kdl/tests/solvertest.cpp [hunk 1]
```diff
@@ -965,8 +965,8 @@ void SolverTest::FdSolverDevelopmentTest()
         {0.,0.199,-0.553,0.788,-0.615,0.162,-0.816},
         {0.,-0.980,-0.112,-0.392,-0.536,-0.803,-0.436},
         {1.,0.,0.825,0.475,0.578,-0.573,0.381}};
-    for ( int i=0; i<6; i++ ) {
-        for ( int j=0; j<nj; j++ ) {
+    for ( unsigned int i=0; i<6; i++ ) {
```
Comment (MatthijsBurgh) [inline orocos_kdl/tests/solvertest.cpp:968]:
```suggestion
    for (unsigned int i=0; i<6; i++ ) {
```

File: orocos_kdl/tests/solvertest.cpp [hunk 2]
```diff
@@ -965,8 +965,8 @@ void SolverTest::FdSolverDevelopmentTest()
...
+        for ( unsigned int j=0; j<nj; j++ ) {
```
Comment (MatthijsBurgh) [inline orocos_kdl/tests/solvertest.cpp:969]:
```suggestion
        for (unsigned int j=0; j<nj; ++j) {
```

File: orocos_kdl/tests/solvertest.cpp [hunk 3]
```diff
@@ -1007,8 +1007,8 @@ void SolverTest::FdSolverDevelopmentTest()
         {0.0638,0.0003,0.1150,-0.0201,1.1234,0.0029,0.0955},
         {-0.0054,-0.0242,-0.0243,0.0087,0.0029,1.1425,0},
         {0.0381,0.0265,0.0814,-0.0291,0.0955,0,1.1000}};
-    for ( int i=0; i<nj; i++ ) {
-        for ( int j=0; j<nj; j++ ) {
+    for ( unsigned int i=0; i<nj; i++ ) {
```
Comment (MatthijsBurgh) [inline orocos_kdl/tests/solvertest.cpp:1010]:
```suggestion
    for (unsigned int i=0; i<nj; ++i ) {
```

File: orocos_kdl/tests/solvertest.cpp [hunk 4]
```diff
@@ -1007,8 +1007,8 @@ void SolverTest::FdSolverDevelopmentTest()
...
+        for ( unsigned int j=0; j<nj; j++ ) {
```
Comment (MatthijsBurgh) [inline orocos_kdl/tests/solvertest.cpp:1011]:
```suggestion
        for (unsigned int j=0; j<nj; ++j) {
```

File: orocos_kdl/tests/solvertest.cpp [hunk 5]
```diff
@@ -1027,7 +1027,7 @@ void SolverTest::FdSolverDevelopmentTest()
     Wrench f_tool(f,n);
     // In local link coordinates
     Wrenches f_ext(ns);
-    for(int i=0;i<ns;i++){
+    for(unsigned int i=0;i<ns;i++){
```
Comment (MatthijsBurgh) [inline orocos_kdl/tests/solvertest.cpp:1030]:
```suggestion
    for(unsigned int i=0; i<ns; ++i) {
```

File: orocos_kdl/tests/solvertest.cpp [hunk 6]
```diff
@@ -1047,14 +1047,14 @@ void SolverTest::FdSolverDevelopmentTest()
     // get acceleration using inverse symmetric matrix times vector
     Eigen::MatrixXd H_eig(nj,nj), L(nj,nj);
     Eigen::VectorXd Tnon_eig(nj), D(nj), r(nj), acc_eig(nj);
-    for(int i=0;i<nj;i++){
+    for(unsigned int i=0;i<nj;i++){
```
Comment (MatthijsBurgh) [inline orocos_kdl/tests/solvertest.cpp:1050]:
```suggestion
    for(unsigned int i=0; i<nj; ++i) {
```

File: orocos_kdl/tests/solvertest.cpp [hunk 7]
```diff
@@ -1047,14 +1047,14 @@ void SolverTest::FdSolverDevelopmentTest()
...
         Tnon_eig(i) =  -Tnoninertial(i);
-        for(int j=0;j<nj;j++){
+        for(unsigned int j=0;j<nj;j++){
```
Comment (MatthijsBurgh) [inline orocos_kdl/tests/solvertest.cpp:1052]:
```suggestion
        for(unsigned int j=0; j<nj; ++j) {
```

File: orocos_kdl/tests/solvertest.cpp [hunk 8]
```diff
@@ -1047,14 +1047,14 @@ void SolverTest::FdSolverDevelopmentTest()
...
             H_eig(i,j) =  H(i,j);
         }
     }
     ldl_solver_eigen(H_eig, Tnon_eig, L, D, r, acc_eig);
-    for(int i=0;i<nj;i++){
+    for(unsigned int i=0;i<nj;i++){
```
Comment (MatthijsBurgh) [inline orocos_kdl/tests/solvertest.cpp:1057]:
```suggestion
    for(unsigned int i=0; i<nj; ++i){
```

File: orocos_kdl/tests/solvertest.cpp [hunk 9]
```diff
@@ -1119,7 +1119,7 @@ void SolverTest::FdSolverConsistencyTest()
     KDL::Wrench f_tool(f,n);
     // In local link coordinates
     KDL::Wrenches f_ext(ns);
-    for(int i=0;i<ns;i++){
+    for(unsigned int i=0;i<ns;i++){
```
Comment (MatthijsBurgh) [inline orocos_kdl/tests/solvertest.cpp:1122]:
```suggestion
    for(unsigned int i=0; i<ns; ++i) {
```

File: orocos_kdl/tests/solvertest.cpp [hunk 10]
```diff
@@ -1139,7 +1139,7 @@ void SolverTest::FdSolverConsistencyTest()
     KDL::ChainIdSolver_RNE IdSolver = KDL::ChainIdSolver_RNE(motomansia10dyn, gravity);
     KDL::JntArray torque(nj);
     IdSolver.CartToJnt(q, qd, qdd, f_ext, torque);
-    for ( int i=0; i<nj; i++ )
+    for ( unsigned int i=0; i<nj; i++ )
     {
```
Comment (MatthijsBurgh) [inline orocos_kdl/tests/solvertest.cpp:1143]:
```suggestion
    for (unsigned int i=0; i<nj; ++i) {
```

File: python_orocos_kdl/PyKDL/pybind11/dynamics.cpp [hunk 11]
```diff
@@ -49,7 +49,7 @@ void init_dynamics(pybind11::module &m)
     {
         int i = std::get<0>(idx);
         int j = std::get<1>(idx);
-        if (i < 0 || i >= jm.rows() || j < 0 || j >= jm.columns())
+        if (i < 0 || (unsigned int)i >= jm.rows() || j < 0 || (unsigned int)j >= jm.columns())
```
Comment (MatthijsBurgh) [inline python_orocos_kdl/PyKDL/pybind11/dynamics.cpp:52]:
```suggestion
        if (i < 0 || static_cast<unsigned int>i >= jm.rows() || j < 0 || static_cast<unsigned int>j >= jm.columns())
```

File: python_orocos_kdl/PyKDL/pybind11/dynamics.cpp [hunk 12]
```diff
@@ -58,7 +58,7 @@ void init_dynamics(pybind11::module &m)
...
```
Comment (MatthijsBurgh) [inline python_orocos_kdl/PyKDL/pybind11/dynamics.cpp:61]:
```suggestion
        if (i < 0 || static_cast<unsigned int>(i) >= jm.rows() || j < 0 || static_cast<unsigned int>(j) >= jm.columns())
```

File: python_orocos_kdl/PyKDL/pybind11/kinfam.cpp [hunk 13]
```diff
@@ -227,15 +227,15 @@ void init_kinfam(pybind11::module &m)
     {
         int i = std::get<0>(idx);
         int j = std::get<1>(idx);
-        if (i < 0 || i > 5 || j < 0 || j >= jac.columns())
+        if (i < 0 || i > 5 || j < 0 || (unsigned int)j >= jac.columns())
```
Comment (MatthijsBurgh) [inline python_orocos_kdl/PyKDL/pybind11/kinfam.cpp:230]:
```suggestion
        if (i < 0 || i > 5 || j < 0 || static_cast<unsigned int>(j) >= jac.columns())
```

File: python_orocos_kdl/PyKDL/pybind11/kinfam.cpp [hunk 14]
```diff
@@ -227,15 +227,15 @@ void init_kinfam(pybind11::module &m)
...
             throw py::index_error("Jacobian index out of range");
         return jac((unsigned int)i, (unsigned int)j);
     });
     jacobian.def("__setitem__", [](Jacobian &jac, std::tuple<int, int> idx, double value)
     {
         int i = std::get<0>(idx);
         int j = std::get<1>(idx);
-        if (i < 0 || i > 5 || j < 0 || j >= jac.columns())
+        if (i < 0 || i > 5 || j < 0 || (unsigned int)j >= jac.columns())
```
Comment (MatthijsBurgh) [inline python_orocos_kdl/PyKDL/pybind11/kinfam.cpp:238]:
```suggestion
        if (i < 0 || i > 5 || j < 0 || static_cast<unsigned int>(j) >= jac.columns())
```

File: python_orocos_kdl/PyKDL/pybind11/kinfam.cpp [hunk 15]
```diff
@@ -265,14 +265,14 @@ void init_kinfam(pybind11::module &m)
     jnt_array.def("resize", &JntArray::resize);
     jnt_array.def("__getitem__", [](const JntArray &ja, int i)
     {
-        if (i < 0 || i >= ja.rows())
+        if (i < 0 || (unsigned int)i >= ja.rows())
```
Comment (MatthijsBurgh) [inline python_orocos_kdl/PyKDL/pybind11/kinfam.cpp:268]:
```suggestion
        if (i < 0 || static_cast<unsigned int>(i) >= ja.rows())
```

File: python_orocos_kdl/PyKDL/pybind11/kinfam.cpp [hunk 16]
```diff
@@ -265,14 +265,14 @@ void init_kinfam(pybind11::module &m)
...
             throw py::index_error("JntArray index out of range");
 
         return ja(i);
     });
     jnt_array.def("__setitem__", [](JntArray &ja, int i, double value)
     {
-        if (i < 0 || i >= ja.rows())
+        if (i < 0 || (unsigned int)i >= ja.rows())
```
Comment (MatthijsBurgh) [inline python_orocos_kdl/PyKDL/pybind11/kinfam.cpp:275]:
```suggestion
        if (i < 0 || static_cast<unsigned int>(i) >= ja.rows())
```

File: orocos_kdl/src/chainiksolverpos_nr.cpp [hunk 17]
```diff
@@ -47,24 +47,24 @@ namespace KDL
...
-                // we chose to continue if the child solver returned a positive
-                // "error", which may simply indicate a degraded solution
-                Add(q_out,delta_q,q_out);
-                if(Equal(delta_twist,Twist::Zero(),eps))
-                    // converged, but possibly with a degraded solution
-                    return (rc > E_NOERROR ? E_DEGRADED : E_NOERROR);
-            }
-            return (error = E_MAX_ITERATIONS_EXCEEDED);        // failed to converge
+        unsigned int i;
+        for(i=0;i<maxiter;i++){
```
Comment (MatthijsBurgh) [inline orocos_kdl/src/chainiksolverpos_nr.cpp:53]:
```suggestion
        for(i=0; i<maxiter; ++i) {
```

File: orocos_kdl/src/chainiksolverpos_nr.cpp [hunk 18]
```diff
@@ -47,24 +47,24 @@ namespace KDL
...
+            if (E_NOERROR > fksolver.JntToCart(q_out,f) )
```
Comment (MatthijsBurgh) [inline orocos_kdl/src/chainiksolverpos_nr.cpp:54]:
```suggestion
            if (E_NOERROR > fksolver.JntToCart(q_out,f))
```

File: orocos_kdl/src/chainiksolverpos_nr.cpp [hunk 19]
```diff
@@ -47,24 +47,24 @@ namespace KDL
...
+            if (E_NOERROR > rc)
+                return (error = E_IKSOLVER_FAILED);
+            // we chose to continue if the child solver returned a positive
+            // "error", which may simply indicate a degraded solution
+            Add(q_out,delta_q,q_out);
+            if(Equal(delta_twist,Twist::Zero(),eps))
+                // converged, but possibly with a degraded solution
+                return (rc > E_NOERROR ? E_DEGRADED : E_NOERROR);
+        }
+        return (error = E_MAX_ITERATIONS_EXCEEDED);        // failed to converge
```
Comment (MatthijsBurgh) [inline orocos_kdl/src/chainiksolverpos_nr.cpp:67]:
```suggestion
        return (error = E_MAX_ITERATIONS_EXCEEDED); // failed to converge
```

File: orocos_kdl/src/chainiksolverpos_nr_jl.cpp [hunk 20]
```diff
@@ -67,37 +67,37 @@ namespace KDL
...
-            q_out = q_init;
+        q_out = q_init;
 
-            unsigned int i;
-            for(i=0;i<maxiter;i++){
-                if ( fksolver.JntToCart(q_out,f) < 0)
-                    return (error = E_FKSOLVERPOS_FAILED);
-                delta_twist = diff(f,p_in);
+        unsigned int i;
+        for(i=0;i<maxiter;i++){
```
Comment (MatthijsBurgh) [inline orocos_kdl/src/chainiksolverpos_nr_jl.cpp:73]:
```suggestion
        for(i=0; i<maxiter; ++i) {
```

File: orocos_kdl/src/chainiksolverpos_nr_jl.cpp [hunk 21]
```diff
@@ -67,37 +67,37 @@ namespace KDL
...
+            if ( fksolver.JntToCart(q_out,f) < 0)
```
Comment (MatthijsBurgh) [inline orocos_kdl/src/chainiksolverpos_nr_jl.cpp:74]:
```suggestion
            if (fksolver.JntToCart(q_out,f) < 0)
```

File: orocos_kdl/src/chainiksolverpos_nr_jl.cpp [hunk 22]
```diff
@@ -67,37 +67,37 @@ namespace KDL
...
 
-        if(Equal(delta_twist,Twist::Zero(),eps))
-            break;
+            if(Equal(delta_twist,Twist::Zero(),eps))
+                break;
 
-        if ( iksolver.CartToJnt(q_out,delta_twist,delta_q) < 0)
-            return (error = E_IKSOLVERVEL_FAILED);
-                Add(q_out,delta_q,q_out);
+            if ( iksolver.CartToJnt(q_out,delta_twist,delta_q) < 0)
```
Comment (MatthijsBurgh) [inline orocos_kdl/src/chainiksolverpos_nr_jl.cpp:81]:
```suggestion
            if (iksolver.CartToJnt(q_out,delta_twist,delta_q) < 0)
```

File: orocos_kdl/src/chainiksolverpos_nr_jl.cpp [hunk 23]
```diff
@@ -67,37 +67,37 @@ namespace KDL
...
+                return (error = E_IKSOLVERVEL_FAILED);
+            Add(q_out,delta_q,q_out);
 
-                for(unsigned int j=0; j<q_min.rows(); j++) {
-                  if(q_out(j) < q_min(j))
-                    q_out(j) = q_min(j);
-                }
+            for(unsigned int j=0; j<q_min.rows(); j++) {
```
Comment (MatthijsBurgh) [inline orocos_kdl/src/chainiksolverpos_nr_jl.cpp:85]:
```suggestion
            for(unsigned int j=0; j<q_min.rows(); ++j) {
```

File: orocos_kdl/src/chainiksolverpos_nr_jl.cpp [hunk 24]
```diff
@@ -67,37 +67,37 @@ namespace KDL
...
+                if(q_out(j) < q_min(j))
```
Comment (MatthijsBurgh) [inline orocos_kdl/src/chainiksolverpos_nr_jl.cpp:86]:
```suggestion
                if (q_out(j) < q_min(j))
```

File: orocos_kdl/src/chainiksolverpos_nr_jl.cpp [hunk 25]
```diff
@@ -67,37 +67,37 @@ namespace KDL
...
+                q_out(j) = q_min(j);
+            }
 
 
-                for(unsigned int j=0; j<q_max.rows(); j++) {
-                    if(q_out(j) > q_max(j))
-                      q_out(j) = q_max(j);
-                }
+            for(unsigned int j=0; j<q_max.rows(); j++) {
```
Comment (MatthijsBurgh) [inline orocos_kdl/src/chainiksolverpos_nr_jl.cpp:91]:
```suggestion
            for(unsigned int j=0; j<q_max.rows(); ++j) {
```

File: orocos_kdl/src/chainiksolverpos_nr_jl.cpp [hunk 26]
```diff
@@ -67,37 +67,37 @@ namespace KDL
...
+                if(q_out(j) > q_max(j))
+                    q_out(j) = q_max(j);
             }
+        }
 
-            if(i!=maxiter)
-                return (error = E_NOERROR);
-            else
-                return (error = E_MAX_ITERATIONS_EXCEEDED);
+        if(i!=maxiter)
```
Comment (MatthijsBurgh) [inline orocos_kdl/src/chainiksolverpos_nr_jl.cpp:97]:
```suggestion
        if(i != maxiter)
```

File: orocos_kdl/src/jntspaceinertiamatrix.cpp [hunk 27]
```diff
@@ -115,8 +115,8 @@ namespace KDL
         return src1.data.isApprox(src2.data,eps);
     }
 
-    bool operator==(const JntSpaceInertiaMatrix& src1,const JntSpaceInertiaMatrix& src2){return Equal(src1,src2);};
-    //bool operator!=(const JntSpaceInertiaMatrix& src1,const JntSpaceInertiaMatrix& src2){return Equal(src1,src2);};
+    bool operator==(const JntSpaceInertiaMatrix& src1,const JntSpaceInertiaMatrix& src2){return Equal(src1,src2);}
+    //bool operator!=(const JntSpaceInertiaMatrix& src1,const JntSpaceInertiaMatrix& src2){return Equal(src1,src2);}
```
Comment (MatthijsBurgh) [inline orocos_kdl/src/jntspaceinertiamatrix.cpp:119]:
```suggestion
    bool operator==(const JntSpaceInertiaMatrix& src1,const JntSpaceInertiaMatrix& src2){ return Equal(src1, src2); }
    //bool operator!=(const JntSpaceInertiaMatrix& src1,const JntSpaceInertiaMatrix& src2){ return !Equal(src1, src2); }
```

**Comment by meyerj** [comment]
> Travis file is good to go. For the code files, I have added a whole set of code suggestions mostly related to styling.

The style is currently inconsistent for the whole project (with/without spaces after the opening and before the closing parenthesis, C-style casts vs. `static_cast`, etc.). Although I agree that your suggestions in terms of style and that `static_cast` is preferable, I would not apply that now for just the changed lines, and rather use a tool like [clang-format](https://clang.llvm.org/docs/ClangFormat.html) or [uncrustify](https://github.com/uncrustify/uncrustify) to fix the problem systematically for the whole repository in a follow-up PR.

**Comment by MatthijsBurgh** [comment]
@meyerj Ok.

So are you giving me a card blance to go ahead and fix the styling of the entire repo? Which will affect the git blame a lot.

**Comment by meyerj** [comment]
> @meyerj Ok.
> 
> So are you giving me a card blance to go ahead and fix the styling of the entire repo? Which will affect the git blame a lot.

True. I leave that to @smits.

But as a first step we would have to decide on a code style to apply for this project. Using 4 spaces for indentation and spaces inside parenthesis was quite common for older [Orocos](https://www.orocos.org/) code. Other influences are the [ROS C++ style guide](http://wiki.ros.org/CppStyleGuide) or [ROS 2 C++ code style](https://index.ros.org/doc/ros2/Contributing/Code-Style-Language-Versions/#id1), which is derived from, but not compatible to [Google C++ style](https://google.github.io/styleguide/cppguide.html). Without having made that decision first, it does not make much sense to apply style-changes in a few places only based on personal preference, and consistency with existing code is more important.

**Pull request merged**